app/models/publication.rb

```diff
@@ -159,6 +159,13 @@ def self.preferred_versions
       .where(%{NOT EXISTS (SELECT * FROM activity_insight_oa_files WHERE activity_insight_oa_files.publication_id = publications.id AND publications.preferred_version = activity_insight_oa_files.version)})
       .where(%{NOT EXISTS (SELECT * FROM activity_insight_oa_files WHERE activity_insight_oa_files.publication_id = publications.id AND activity_insight_oa_files.version IS NULL)})
   }
+  scope :wrong_file_version, -> {
+    activity_insight_oa_publication
+      .where.not(preferred_version: nil)
+      .where(%{NOT EXISTS (SELECT * FROM activity_insight_oa_files WHERE activity_insight_oa_files.publication_id = publications.id AND activity_insight_oa_files.version = 'unknown')})
+      .where(%{NOT EXISTS (SELECT * FROM activity_insight_oa_files WHERE activity_insight_oa_files.publication_id = publications.id AND publications.preferred_version = activity_insight_oa_files.version)})
```

Comment:
Just a heads-up that this is going to get more complicated when we get to https://github.com/psu-libraries/researcher-metadata/issues/789, but we'll cross that bridge when we come to it.